com.aobuchow.themes.spectrum.preferences/src/com/aobuchow/themes/spectrum/preferences/ColorManager.java

```diff
@@ -56,6 +98,44 @@ public static void setStyledTextColoring(StyledText colorScheme, ColorRegistry c
 		accentStyle.foreground = ColorUtils.useReadableForegroundColor(accentColor, white, black);
 		accentStyle.background = accentColor;
 		colorScheme.setStyleRange(accentStyle);
+		
+		white.dispose();
+		black.dispose();
+	}
+	
+	public void dispose() {
+		accentColor.dispose();
+		baseColor.dispose();
+		backgroundColor.dispose();
+	}
+
+	private void updateGitColors() {
+		ColorHSL uncommittedChangeBackground = new ColorHSL(backgroundColor).modifyProperty(MODIFICATION.INCREASE,
+				HSL_PROPERTY.LUMINANCE, BOUND_BEHAVIOR.REVERSE, 0.1f);
+		uncommittedChangeBackground = uncommittedChangeBackground.modifyProperty(MODIFICATION.INCREASE,
+				HSL_PROPERTY.SATURATION, BOUND_BEHAVIOR.LIMIT, 0.1f);
+		uncommittedChangeBackground = uncommittedChangeBackground.modifyProperty(MODIFICATION.INCREASE,
+				HSL_PROPERTY.HUE, BOUND_BEHAVIOR.CYCLE, 5f);
+
+		Color uncommittedChangeForeground;
+		if (uncommittedChangeBackground.getPerceivedLuminance() >= 127.5f) {
+			uncommittedChangeForeground = new Color(display, 0, 0, 0);
+		} else {
+			uncommittedChangeForeground = new Color(display, 255, 255, 255);
+		}
+		setColorPreference("org.eclipse.egit.ui.UncommittedChangeBackgroundColor",
+				uncommittedChangeBackground.getColor());
+		setColorPreference("org.eclipse.egit.ui.UncommittedChangeForegroundColor",
+				uncommittedChangeForeground);
+		
+		uncommittedChangeBackground.dispose();
+		uncommittedChangeForeground.dispose();
+	}
+
+	private void setColorPreference(String preferenceKey, Color color) {
+		// We can't use PlatformUI.getPreferenceStore() as it won't affect preferences for plugins such as EGit
```

Comment:
@ingomohr unfortunately I'm also currently unable to find a workaround for using this deprecated API. The issue is that we need to be able to override egit's preferences as well (and probably other preferences in the future).

Reply:
@AObuchow From what I see in the source code of ``eclipse.ui.workbench`` I too don't see a solution to use some actual API.
So, maybe your current solution is the one to go with. :)

Reply:
Sounds good. Ideally, I should discuss this further with the upstream community. However, they might just suggest not changing other plugin's preferences :P